crates/eips/src/eip2718.rs

```diff
@@ -178,6 +179,7 @@ pub trait Decodable2718: Sized {
 /// over the accepted transaction types.
 ///
 /// [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
+#[auto_impl(keep_default_for(&))]
```

Comment:
this must go on the function that returns `Self`

and here we need to add just auto_impl(&)